looping-through-arrays.js

```diff
@@ -0,0 +1,5 @@
+var pets = ['cat', 'dog', 'rat'];
+for(i=0; i<pets.length; i++){
```

Comment:
cool. an optimization here is...

```
var cached = pets.length
for(var i=0;i<cached;i++){
```

theres also something important to remember here. if you don't specify 'var' in your loop you're declaring `window.i` which can cause all types of havoc.